src/modules/hip/kernel/lens_correction.hpp

```diff
@@ -0,0 +1,119 @@
+        invMat->f1[7] = (mat->f1[6] * mat->f1[1] - mat->f1[0] * mat->f1[7]) * invDet;
+        invMat->f1[8] = (mat->f1[0] * mat->f1[4] - mat->f1[3] * mat->f1[1]) * invDet;
+    }
+}
+
+__global__ void compute_remap_tables_hip_tensor(float *rowRemapTable,
+                                                float *colRemapTable,
+                                                d_float9 *cameraMatrixTensor,
+                                                d_float9 *inverseMatrixTensor,
+                                                d_float8 *distortionCoeffsTensor,
+                                                d_float9 *newCameraMatrixTensor,
+                                                uint2 remapTableStridesNH,
+                                                RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if ((id_y >= roiTensorPtrSrc[id_z].xywhROI.roiHeight) || (id_x >= roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+    {
+        return;
+    }
+
+    int height = roiTensorPtrSrc[id_z].xywhROI.roiHeight;
+    int width = roiTensorPtrSrc[id_z].xywhROI.roiWidth;
+    d_float9 cameraMatrix = cameraMatrixTensor[id_z];
+    d_float9 newCameraMatrix = newCameraMatrixTensor[id_z];
+    d_float9 inverseMatrix = inverseMatrixTensor[id_z];
+    d_float8 distortionCoeffs = distortionCoeffsTensor[id_z];
+
+    // Get radial and tangential distortion coefficients
+    float rCoeff[6] = { distortionCoeffs.f1[0], distortionCoeffs.f1[1], distortionCoeffs.f1[4], distortionCoeffs.f1[5], distortionCoeffs.f1[6], distortionCoeffs.f1[7] };
+    float tCoeff[2] = { distortionCoeffs.f1[2], distortionCoeffs.f1[3] };
+
+    // Get the focal length and principal point of the camera
+    float u0 = cameraMatrix.f1[2],  v0 = cameraMatrix.f1[5];
+    float fx = cameraMatrix.f1[0],  fy = cameraMatrix.f1[4];
+
+    float *rowRemapTableTemp = rowRemapTable + id_z * remapTableStridesNH.x + id_y * remapTableStridesNH.y + id_x;
+    float *colRemapTableTemp = colRemapTable + id_z * remapTableStridesNH.x + id_y * remapTableStridesNH.y + id_x;
+    float xCamera = id_y * inverseMatrix.f1[1] + inverseMatrix.f1[2] + id_x * inverseMatrix.f1[0];
+    float yCamera = id_y * inverseMatrix.f1[4] + inverseMatrix.f1[5] + id_x * inverseMatrix.f1[3];
+    float zCamera = id_y * inverseMatrix.f1[7] + inverseMatrix.f1[8] + id_x * inverseMatrix.f1[6];
+    float z = 1./zCamera, x = xCamera * z, y = yCamera * z;
+    float xSquare = x * x, ySquare = y * y;
+    float r2 = xSquare + ySquare, xyMul2 = 2 * x * y;
+    float kr = (1 + ((rCoeff[2] * r2 + rCoeff[1]) * r2 + rCoeff[0]) * r2) / (1 + ((rCoeff[5] * r2 + rCoeff[4]) * r2 + rCoeff[3]) *r2);
+    float colLoc = fx * (x * kr + tCoeff[0] * xyMul2 + tCoeff[1] * (r2 + 2 * xSquare)) + u0;
```

Comment:
Why can't this kernel be vectorized along x? All these arithmetic computations could be float4 computes too